yorkie/src/main/kotlin/dev/yorkie/document/crdt/RgaTreeList.kt

```diff
@@ -0,0 +1,238 @@
+        val newNode = RgaTreeListNode.createAfter(prevNode, value)
+        if (prevNode == last) {
+            last = newNode
+        }
+        nodeMapByIndex.insertAfter(prevNode.value, newNode.value)
+        nodeMapByCreatedAt[newNode.createdAt.toIDString()] = newNode
+    }
+
+    /**
+     * Returns the node by the given [createdAt] and [executedAt].
+     * It passes through nodes created after [executedAt] from the
+     * given node and returns the next node.
+     */
+    private fun findNextBeforeExecutedAt(
+        createdAt: TimeTicket,
+        executedAt: TimeTicket,
+    ): RgaTreeListNode {
+        var node = nodeMapByCreatedAt[createdAt.toIDString()]
+            ?: error("can't find the given node createdAt: ${createdAt.toIDString()}")
+
+        var nodeNext = node.next
+        while (nodeNext != null && executedAt < nodeNext.positionedAt) {
+            node = nodeNext
+            nodeNext = node.next
+        }
+        return node
+    }
+
+    /**
+     * Moves the given [createdAt] element after the [prevCreatedAt] element.
+     */
+    fun moveAfter(
+        prevCreatedAt: TimeTicket,
+        createdAt: TimeTicket,
+        executedAt: TimeTicket,
+    ) {
+        val prevNode = nodeMapByCreatedAt[prevCreatedAt.toIDString()]
+            ?: error("can't find the given node createdAt: ${prevCreatedAt.toIDString()}")
+        val node = nodeMapByCreatedAt[createdAt.toIDString()]
+            ?: error("can't find the given node createdAt: ${createdAt.toIDString()}")
+
+        if (prevNode != node &&
+            (node.value.movedAt == null || checkNotNull(node.value.movedAt) < executedAt)
+        ) {
+            release(node)
+            insertAfter(prevNode.createdAt, node.value, executedAt)
+            node.value.movedAt = executedAt
+        }
+    }
+
+    // NOTE(7hong13): original comment from JS SDK is as follows:
+    // `purge` physically purges child element.
+    /**
+     * Physically purges element.
+     */
+    fun purge(element: CrdtElement) {
+        val node = nodeMapByCreatedAt[element.createdAt.toIDString()]
+            ?: error("can't find the given createdAt: ${element.createdAt.toIDString()}")
+        release(node)
+    }
+
+    private fun release(node: RgaTreeListNode) {
+        if (last == node) {
+            last = requireNotNull(node.prev)
+        }
+        node.release()
+        nodeMapByIndex.delete(node.value)
+        nodeMapByCreatedAt.remove(node.value.createdAt.toIDString())
+    }
+
+    // NOTE(7hong13): original comment from JS SDK is as follows:
+    // `get` returns the element of the given index.
+    /**
+     * Returns the element of the given key [createdAt].
+     */
+    fun get(createdAt: TimeTicket): CrdtElement? {
+        return nodeMapByCreatedAt[createdAt.toIDString()]?.value
+    }
+
+    /**
+     * Returns the key based on [createdAt] value of the node.
+     */
+    fun keyOf(createdAt: TimeTicket): String? {
```

Comment:
need a new function name to reveal its usage properly

Reply:
How about looking up what terminology the JSONPath library uses?

https://github.com/json-path/JsonPath